Patterns/LayeredProject.md

```diff
@@ -0,0 +1,100 @@
+
+### Additional folders
+
+Solution folders can be used to group specific projects together. If for example a gateway uses multiple project, group these projects in a solution folder describing the gateway. 
+
+### Shorter project names
+
+When grouping projects in solution folders, the names of these projects can be shortened to simplify their naming. Instead of `{Customer}.{Project}.Gateway.{External API}.{External API project}.csproj` the name `{External API}.{External API project}.csproj` can be used.
+
+### Project folders and namespaces
+
+Folders within projects should group similar objects (to a certain degree), and should be named what seems logical. Group converters in `Converters` and controllers in `Controllers`. Add more layers of folders when the folder becomes too big, like adding `Request` and `Response` folders in `Models` to separate request and response models. Group interfaces in `Abstractions`. Group extension (methods) in `Extensions`.
+
+The namespace of a file should always be comprised of its compound folder name, like `Contoso.CalendarHub.Api.Models.Request`.
+
+### Example
+
+- Contoso.CalendarHub.sln
+  - Contoso.CalendarHub.Api.csproj
+    - `Controllers`
+    - `JsonConverters`
+    - `Models`
+        - `Request`
+        - `Response`
+  - Contoso.CalendarHub.Api.Tests.csproj
+    - `LoginControllerTests`
+    - `UserControllerTests`
+  - Contoso.CalendarHub.Business.csproj
+    - `Abstractions`
+        - `Calendars`
+        - `Users`
+    - `Services`
+        - `Calendar`
+        - `User`
+  - Contoso.CalendarHub.Business.Tests.csproj
+    - `UserServiceTests`
+    - `CalendarServiceTest`
+  - Contoso.CalendarHub.Core.csproj
+    - `Attributes`
+    - `Extensions`
+    - `Enums`
+    - `Models`
+    - `Resolvers`
+  - Contoso.CalendarHub.Cms.csproj
+  - Contoso.CalendarHub.Data.Users.csproj
+  - Contoso.CalendarHub.Data.Invitations.csproj
+  - Contoso.CalendarHub.Functions.csproj
+    - `Attributes`
+    - `Constants`
+    - `Extensions`
+  - Contoso.CalendarHub.Gateway.Adyen.csproj
+    - `Models`
+  - Contoso.CalendarHub.Gateway.GoogleCalendar.csproj
+    - `Models`
+    - `Helpers`
+  - Contoso.CalendarHub.Gateway.GoogleCalendar.Tests.csproj
+  - Contoso.CalendarHub.Gateway.OutlookCalendar.csproj
+    - `Models`
+  - Contoso.CalendarHub.Gateway.OutlookCalendar.Tests.csproj
+
+## Inter-project data exchange and isolation
+
+When dividing a solution into several distinct projects it is important to maintain a level of isolation between these projects. Keeping these projects reasonably isolated helps to control the impact of changes; a change is one project should not force a lot of changes in other projects. Isolation will prevent developers from having to notify users of the API of your solution that some outbound data changes due to modification in a database column. Keep in mind to not overdo this, implementation details will always tend to spread a little bit to other parts of the solution.
+
```

Comment:
Mention the Single Responsibility Principle? _A class or module should have one, and only one **reason to change**._ So when a minor change in one project/module forces (many) other changes in other projects/modules, you know bad stuff has happened!

Reply:
The single responsibility principle does not shield you from spreading your implementation details to everywhere. An object that does only one thing can still be `new`ed everywhere and when it's API surface changes force you to change your entire application. So, its a good thing, but it is not the only answer to isolation.